<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="environment" default="garage"/>
  <node name="metrics_run" pkg="autonomous_tidy_up" output="screen" type="metrics_node" args="_dir_arq:=$(find autonomous_tidy_up)/logs/$(arg environment)">
  </node>

  <env name="LC_NUMERIC" value="C" />

  <!-- Load definition of pregrasp motion into play_motion namespace -->
  <rosparam command="load" file="$(find tiago_pick_demo)/config/pick_motions.yaml" /> 

  <include file="$(find tiago_moveit_tutorial)/launch/octomap_tiago.launch"></include> <!--start rviz-->

  <rosparam command="load" file="$(find autonomous_tidy_up)/config/tiago_custom_motions.yaml" /> 

  <!-- exploration_and_manipulation -->
  <node name="autonomous_tidy_up_run" pkg="autonomous_tidy_up" output="screen" type="autonomous_tidy_up_node">
    <param name="ee_group_name" value="gripper"/>
    <param name="planning_group_name" value="arm_torso"/>
    <param name="dir_arq" value="$(find autonomous_tidy_up)/logs/$(arg environment)"/>
    <rosparam command="load" file="$(find moveit_grasps)/config_robot/tiago_grasp_data.yaml"/>
    <rosparam command="load" file="$(find moveit_grasps)/test/config/moveit_grasps_config.yaml"/>
    <remap from="/autonomous_tidy_up_run/grasping_planning_scene" to="/planning_scene" />
  </node>

  <!-- Planning Functionality -->
  <arg name="planner" default="ompl" />
  <include ns="autonomous_tidy_up" file="$(find tiago_moveit_config)/launch/planning_pipeline.launch.xml">
    <arg name="pipeline" value="$(arg planner)" />
  </include>

</launch>